webdriver/tests/get_element_shadow_root/get.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+from tests.support.asserts import assert_error, assert_success
+
+
+def get_shadow_root(session, shadow_id):
+    return session.transport.send(
+        "GET", "session/{session_id}/element/{shadow_id}/shadow".format(
+            session_id=session.session_id,
+            shadow_id=shadow_id))
+
+
+def test_no_top_browsing_context(session, closed_window):
+    original_handle, element = closed_window
+    response = get_shadow_root(session, element.id)
+    assert_error(response, "no such window")
+    response = get_shadow_root(session, "foo")
+    assert_error(response, "no such window")
+
+    session.window_handle = original_handle
+    response = get_shadow_root(session, element.id)
+    assert_error(response, "no such element")
+
+
+def test_no_browsing_context(session, closed_frame):
+    response = get_shadow_root(session, "foo")
+    assert_error(response, "no such window")
+
+
+def test_element_not_found(session):
+    result = get_shadow_root(session, "foo")
+    assert_error(result, "no such element")
+
+
+def test_element_stale(session, get_checkbox_dom):
+    session.url = get_checkbox_dom
+    element = session.find.css("custom-checkbox-element", all=False)
+    session.refresh()
+
+    result = get_shadow_root(session, element.id)
+    assert_error(result, "stale element reference")
+
+
+def test_get_shadow_root(session, get_checkbox_dom):
+    session.url = get_checkbox_dom
+    custom_element = session.find.css("custom-checkbox-element", all=False)
+    response = get_shadow_root(session, custom_element)
+    assert_success(response)
```

Comment:
Please add an assertion that checks for the correct shadow root.

Reply:
Assertion added in 8ef520e. Slightly convoluted, in that we compare the `host` of the shadow root to the element we're requesting the shadow root for, but since we don't have a way to pass a shadow root back via JS to compare, that's the best approach I can think of. Feel free to make another suggestion.

Reply:
Hm, shouldn't we able to serialize a shadow root element as retrieved by `Execute Script` similar to a [window or frame object](https://searchfox.org/mozilla-central/source/testing/web-platform/tests/webdriver/tests/execute_script/json_serialize_windowproxy.py)?

Reply:
Maybe, but that's not how it's specified in the spec language. Plus a Shadow Root "element" isn't an element, it's a different thing. We can, in theory, serialize the window and frame proxies, but no implementation [actually does](https://wpt.fyi/results/webdriver/tests/execute_script/json_serialize_windowproxy.py?label=master&label=experimental&aligned). That would also require changes in the `Execute Script` spec language to support it, which wasn't part of this effort. If, in the future, we decide to make this serializable via `Execute Script`, we can update these tests to utilize that mechanism.

Reply:
So I mixed that up. Sorry. In that case lets resolve it.